extension/source/background/KeyringController.ts

```diff
@@ -154,6 +159,30 @@ export default class KeyringController {
 
       await this.keyring.update({ wallets: newWallets });
     },
+
+    /**
+     * Recovers an existing BLS wallet and adds the new
+     * recovered wallet to the keyring
+     * @param recoveryWalletAddress Smart contract address
+     * of wallet being recovered
+     * @param recoverySaltHash Salt used to set the recovery
+     * hash on the wallet that is being recovered
+     * @param signerWalletPrivateKey the private key of the wallet that is used
+     * to generate the recovery hash. This wallet will sign the 'recoverWallet'
+     * request.
+     */
+    addRecoveryWallet: async ({
+      params: [recoveryWalletAddress, recoverySaltHash, signerWalletPrivateKey],
```

Comment:
@kautukkundan There's one thing I wanted to raise that I'm doing differently than what we discussed on our call today.

I'm now accepting a `signerWalletPrivatekey` as a param here.  That's because, based off our convo earlier, you are going to generate a wallet on the fly to be used for the recovery.  This way, you can just pass that wallet's PK to this function.